pkg/prometheus/operator.go

```diff
@@ -982,13 +982,24 @@ func (c *Operator) loadBasicAuthSecrets(mons map[string]*monitoringv1.ServiceMon
 	for i, remote := range remoteWrites {
 		if remote.BasicAuth != nil {
 			if credentials, err := loadBasicAuthSecret(remote.BasicAuth, s); err != nil {
-				return nil, fmt.Errorf("Could not generate basicAuth for remote_write config %d. %s", i, err)
+				return nil, fmt.Errorf("could not generate basicAuth for remote_write config %d. %s", i, err)
 			} else {
 				secrets[fmt.Sprintf("remoteWrite/%d", i)] = credentials
 			}
 		}
 	}
 
+	// load apiserver basic auth secret
+	if apiserverConfig != nil && apiserverConfig.BasicAuth != nil {
+		if apiserverConfig.BasicAuth != nil {
```

Comment:
Haven't you already checked this in the line above in the second expression of the if clause?

Reply:
Sorry, It may be a mistake.